.github/workflows/tpch_benchmark.yml

```diff
@@ -0,0 +1,308 @@
+
+      - name: Build spice
+        run: make -C bin/spice
+
+      - name: make spiced executable
+        if: matrix.target_os != 'windows'
+        run: |
+          mv target/release/spiced spiced
+          chmod +x spiced
+
+      - name: make spice executable
+        if: matrix.target_os != 'windows'
+        run: |
+          mv target/release/spice spice
+          chmod +x spice
+
+      - name: Save spice artifact
+        uses: actions/upload-artifact@v4
+        if: matrix.target_os != 'windows'
+        with:
+          name: build_${{ matrix.target_os }}_${{ matrix.target_arch }}
+          path: |
+            spice
+            spiced
+
+  tpc_h_benchmark:
+    name: "Run TPC-H Benchmark Test ${{ matrix.target_os }}-${{ matrix.target_arch }}"
+    runs-on: ${{ matrix.runner }}
+    #needs: build
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Linux x64
+            runner: ubuntu-latest
+            target_os: linux
+            target_arch: x86_64
+            target_arch_go: amd64
+          # - name: macOS aarch64 (Apple Silicon)
+          #   runner: macos-14
+          #   target_os: darwin
+          #   target_arch: aarch64
+          #   target_arch_go: arm64
+          # - name: macOS x64 (Intel)
+          #   runner: macos-12
+          #   target_os: darwin
+          #   target_arch: x86_64
+          #   target_arch_go: amd64
+          # - name: Windows x64
+          #   runner: rust-windows-x64
+          #   target_os: windows
+          #   target_arch: x86_64
+          #   target_arch_go: amd64
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install PostgreSQL (Linux)
+        if: matrix.target_os == 'linux'
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y postgresql
+          sudo service postgresql start
+          sleep 5
+          sudo -u postgres psql -tc "SELECT 1 FROM pg_roles WHERE rolname='postgres'" | grep -q 1 || sudo -u postgres createuser -s postgres
+          sudo -u postgres psql -c "ALTER USER postgres PASSWORD 'postgres';"
+          sudo -u postgres createdb tpch
+
+      - name: Install PostgreSQL (MacOS)
+        if: matrix.target_os == 'darwin'
+        run: |
+          brew install postgresql
+          brew services start postgresql
+          sleep 5
+          createuser -s postgres
+          psql -d postgres -c "ALTER USER postgres PASSWORD 'postgres';"
+          createdb tpch
+
+      - name: Wait for PostgreSQL to start
+        run: sleep 10
+
+      - name: Check PostgreSQL
+        env:
+          PGPASSWORD: postgres
+        run: psql -h localhost -U postgres -c 'SELECT version();'
+
+      - name: Generate TPC-H Benchmark data
+        run: |
+          git clone https://github.com/gregrahn/tpch-kit.git
+          cd tpch-kit/dbgen
+          make MACHINE=LINUX DATABASE=POSTGRESQL
+          ./dbgen -vf -s 0.1
+
+      - name: Load TPC-H Benchmark data to PostgreSQL
```

Comment:
For TPC-H specific, I think we'll want to have it in a separate script that we can run locally or through Actions.

Reply:
@lukekim - like it, I'll create a script + instructions how to generate / run TPC-H  locally and then update the workflow to rely on the script.

Reply:
The note above ha been addressed:
1. Added https://github.com/spiceai/spiceai/pull/905
2. Updated PR to rely on created script